x-pack/solutions/security/plugins/security_solution/server/endpoint/services/workflow_insights/helpers.ts

```diff
@@ -163,45 +163,65 @@ export function getUniqueInsights(insights: SecurityWorkflowInsight[]): Security
   return Object.values(uniqueInsights);
 }
 
-export const generateTrustedAppsFilter = (insight: SecurityWorkflowInsight): string | undefined => {
-  return insight.remediation.exception_list_items
-    ?.flatMap((item) =>
-      item.entries.map((entry) => {
-        if (!('value' in entry)) return '';
-
-        if (entry.field === 'process.executable.caseless') {
-          return `exception-list-agnostic.attributes.entries.value:"${entry.value}"`;
-        }
-
-        if (
-          entry.field === 'process.code_signature' ||
-          (entry.field === 'process.Ext.code_signature' && typeof entry.value === 'string')
-        ) {
-          const sanitizedValue = (entry.value as string)
-            .replace(/[)(<>}{":\\]/gm, '\\$&')
-            .replace(/\s/gm, '*');
-          return `exception-list-agnostic.attributes.entries.entries.value:(*${sanitizedValue}*)`;
-        }
-
-        return '';
-      })
-    )
-    .filter(Boolean)
-    .join(' AND ');
+export const generateTrustedAppsFilter = (
+  insight: SecurityWorkflowInsight,
+  packagePolicyId?: string
+): string | undefined => {
+  const filterParts =
+    insight.remediation.exception_list_items
+      ?.flatMap((item) =>
+        item.entries.map((entry) => {
+          if (!('value' in entry)) return '';
+
+          if (entry.field === 'process.executable.caseless') {
+            return `exception-list-agnostic.attributes.entries.value:"${entry.value}"`;
+          }
+
+          if (
+            entry.field === 'process.code_signature' ||
+            (entry.field === 'process.Ext.code_signature' && typeof entry.value === 'string')
+          ) {
+            const sanitizedValue = (entry.value as string)
+              .replace(/[)(<>}{":\\]/gm, '\\$&')
+              .replace(/\s/gm, '*');
+            return `exception-list-agnostic.attributes.entries.entries.value:(*${sanitizedValue}*)`;
+          }
+
+          return '';
+        })
+      )
+      .filter(Boolean) || [];
+
+  // Only create a filter if there are valid entries
+  if (filterParts.length) {
+    let combinedFilter = filterParts.join(' AND ');
+    if (packagePolicyId) {
+      const policyFilter = `(exception-list-agnostic.attributes.tags:"policy:${packagePolicyId}" OR exception-list-agnostic.attributes.tags:"policy:all")`;
+      combinedFilter = `${policyFilter} AND ${combinedFilter}`;
+    }
```

Comment:
if there is no `packagePolicyId`, shouldn't we still filter for global artifacts? can it even happen that there is no `packagePolicyId`, or is it optional just because of typing?

Reply:
It’s optional due to the typings, but in our use case, we don’t really expect it to be undefined

Reply:
You’re right, it’s not optional. My mistake, and I appreciate your attention to detail! 😃